shell/browser/api/atom_api_net_log.cc

```diff
@@ -60,13 +85,36 @@ NetLog::NetLog(v8::Isolate* isolate, AtomBrowserContext* browser_context)
 
 NetLog::~NetLog() = default;
 
-v8::Local<v8::Promise> NetLog::StartLogging(mate::Arguments* args) {
-  base::FilePath log_path;
-  if (!args->GetNext(&log_path) || log_path.empty()) {
+v8::Local<v8::Promise> NetLog::StartLogging(base::FilePath log_path,
+                                            mate::Arguments* args) {
+  if (log_path.empty()) {
     args->ThrowError("The first parameter must be a valid string");
     return v8::Local<v8::Promise>();
   }
 
+  net::NetLogCaptureMode capture_mode = net::NetLogCaptureMode::Default();
+  uint64_t max_file_size = network::mojom::NetLogExporter::kUnlimitedFileSize;
+
+  mate::Dictionary dict;
+  if (args->GetNext(&dict)) {
+    v8::Local<v8::Value> capture_mode_v8;
+    if (dict.Get("captureMode", &capture_mode_v8)) {
+      if (!mate::ConvertFromV8(args->isolate(), capture_mode_v8,
```

Comment:
You can convert to target type directly:

```c++
net::NetLogCaptureMode capture_mode;
if (dict.Get("captureMode", &capture_mode)) {
  // ...
}
```

Reply:
The reason not to do this is so that we can produce an error if the option was specified but of the incorrect type, as opposed to silently proceeding with the default value.